repos/system_upgrade/el7toel8/actors/migratesendmail/actor.py

```diff
@@ -0,0 +1,69 @@
+from leapp.actors import Actor
+from leapp.models import SendmailMigrationDecision
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
+from leapp.reporting import Report
+from leapp.libraries.common.reporting import report_generic
+import re
+import ipaddress
+import shutil
+
+BackupSuffix = '.bak'
+
+# false positives blacklist
+rfp = re.compile(r'(^\s*RIPv6:::1\b)|(@\s+\[IPv6:::1\]\s+>)')
+
+rs = re.compile(r'IPv6:[0-9a-fA-F:]*::[0-9a-fA-F:]*')
+
+try:
+    # Python 2: "unicode" is built-in
+    unicode
+except NameError:
+    unicode = str
+
+def uncompress_ipv6(ipv6):
+    addr = unicode(ipv6.split('IPv6:').pop())
+    try:
+        addr = 'IPv6:' + ipaddress.ip_address(addr).exploded
+    except ValueError:
+        addr = ipv6
+    return re.sub(r':0([^:])', r':\1', re.sub(r'0+', r'0', addr))
+
+def check_false_positives(f, l):
+    return f in ["/etc/mail/sendmail.cf", "/etc/mail/submit.cf"] and rfp.search(l) is not None
+
+def sub_ipv6(m):
+    return uncompress_ipv6(m.group(0))
+
+def migrate_file(fn):
+    # make backup
+    shutil.copy2(fn, fn + BackupSuffix)
+    f = open(fn, 'w')
+    for l in open(fn + BackupSuffix):
+        m = rs.search(l)
+        if m is not None and not check_false_positives(fn, l):
+            l = rs.sub(sub_ipv6, l)
+        f.write(l)
+    f.close()
```

Comment:
same as above. the code cannot be written here.

Reply:
Fixed